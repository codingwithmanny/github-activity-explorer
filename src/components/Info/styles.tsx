// IMPORTS
// --------------------------------------------------------
import styled, { keyframes } from 'styled-components';
import color from 'color';
import { Colors, ScreenSizes } from '../../styles/DefaultStyles';

// ANIMATIONS
// --------------------------------------------------------
const fadeUp = keyframes`
    0% {
        transform: translateY(100vh);
    }
    5% {
        opacity: 0;
        transform: translateY(0);
    }
    100% {
        opacity: 1;
        transform: translateY(0);
    }
`;

const fadeDown = keyframes`
    0% {
        opacity: 1;
        transform: translateY(0);
    }
    95% {
        opacity: 0;
        transform: translateY(0);
    }
    100% {
        opacity: 0;
        transform: translateY(100vh);
    }
`;

// TYPES
// --------------------------------------------------------
interface InfoStylesProps {
  show: boolean | null;
}

// MAIN STYLES
// --------------------------------------------------------
/**
 * Main styled the app infor
 * @constant {JSX} styled component
 */
export const InfoStyles = styled.div<InfoStylesProps>`
  display: flex;
  position: fixed;
  justify-content: center;
  align-items: center;
  overflow: scroll;
  z-index: 2000;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  transition: all 250ms ease-in-out 0s;
  background: ${color(Colors.black).alpha(0.7).toString()};
  &:hover,
  &:focus {
    outline: none !important;
  }

  animation: ${(props) =>
      props.show === null ? '' : props.show ? fadeUp : fadeDown}
    400ms ease-in-out 0s forwards;

  ${(props) =>
    props.show === null && `opacity: 0; transform: translateY(100vh);`}

  > div {
    display: block;
    background: ${Colors.white};
    border-radius: 8px;
    width: calc(100% - 100px);
    padding: 20px;
    position: relative;

    button {
      background: ${Colors.deepBlue};
      border: none;
      position: absolute;
      top: 20px;
      right: 20px;
      width: 30px;
      height: 30px;
      border-radius: 30px;
      cursor: pointer;

      svg {
        color: ${Colors.white};
        width: 18px;
        position: absolute;
        top: 0;
        bottom: 0;
        left: 0;
        right: 0;
        margin: auto;
      }

      &:hover,
      &:focus {
        outline: none;
      }
    }

    p {
      color: ${Colors.darkPurple};
      small {
        svg {
          height: 20px;
          margin-right: 5px;
        }
      }
      a {
        margin-bottom: 30px;
        display: block;
        background: ${Colors.grey};
        color: ${Colors.deepBlue};
        line-height: 40px;
        border-radius: 4px;
        padding: 0 15px;
        text-decoration: none;
        transition: all 250ms ease-in-out 0s;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;

        code {
          font-size: 14px;
        }
      }
    }
  }

  @media (min-width: ${ScreenSizes.medium}) {
    > div {
      max-width: 480px;

      button {
        &:focus,
        &:hover {
          box-shadow: 0 0 0 0.2rem rgba(113, 183, 248, 0.3);
        }
      }
      p {
        a {
          &:focus,
          &:hover {
            background: ${color(Colors.grey).darken(0.1).toString()};
            outline: none;
            box-shadow: 0 0 0 0.2rem rgba(113, 183, 248, 0.3);
          }
        }
      }
    }
  }
`;
