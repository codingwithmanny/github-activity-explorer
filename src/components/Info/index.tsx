// IMPORTS
// --------------------------------------------------------
import React, { useEffect } from 'react';
import {
  X,
  Github,
  GitCommit,
  CornerLeftUp,
  User,
} from '@styled-icons/feather';

// Styled Components
import { InfoStyles } from './styles';

// TYPES
// --------------------------------------------------------
interface InfoProps {
  show: boolean | null;
  handleClose?: Function;
}

// MAIN PRESENTATION COMPONENT
// --------------------------------------------------------
/**
 * Info
 * @param {Object} props component props
 * @returns {JSX} React compoennt
 */
const Info: React.FC<InfoProps> = (props) => {
  // State / Props
  const { show, handleClose } = props;

  // Functions
  /**
   * When X button is clicked
   */
  const onClickClose = () => {
    if (handleClose) {
      handleClose();
    }
  };

  /**
   * Handles pressing Escape to leave modal
   */
  useEffect(() => {
    let listener: ((this: Window, ev: KeyboardEvent) => void) | null = null;
    if (show) {
      listener = (event: KeyboardEvent) => {
        if (event.key === 'Escape') {
          onClickClose();
        }
      };
      window.addEventListener('keyup', listener);
    }
    return () => {
      if (listener) {
        window.removeEventListener('keyup', listener);
      }
    };
    // eslint-disable-next-line
  }, [show]);

  // Render
  return (
    <InfoStyles show={show}>
      <div>
        <button onClick={onClickClose}>
          <X strokeWidth={2} />
        </button>
        <p>
          <small>
            <Github strokeWidth={2} />
            GitHub URL
          </small>
        </p>
        <p>
          <a
            href={`${window.GITHUB_REPO_URL}`}
            target="_blank"
            rel="noopener noreferrer"
          >
            <code>{window.GITHUB_REPO_URL}</code>
          </a>
        </p>
        <p>
          <small>
            <GitCommit strokeWidth={2} />
            GitHub COMMIT
          </small>
        </p>
        <p>
          <a
            href={`${window.GITHUB_REPO_URL}/commit/${window.GITHUB_COMMIT}`}
            target="_blank"
            rel="noopener noreferrer"
          >
            <code>{window.GITHUB_COMMIT}</code>
          </a>
        </p>
        <p>
          <small>
            <CornerLeftUp strokeWidth={2} />
            Deployment
          </small>
        </p>
        <p>
          <a
            href={`${window.DEPLOY_URL}`}
            target="_blank"
            rel="noopener noreferrer"
          >
            <code>{window.DEPLOY_ID}</code>
          </a>
        </p>
        <p>
          <small>
            <User strokeWidth={2} />
            GitHub User
          </small>
        </p>
        <p>
          <a
            href={`https://github.com/${window.GITHUB_USER}`}
            target="_blank"
            rel="noopener noreferrer"
          >
            <code>@{window.GITHUB_USER}</code>
          </a>
        </p>
      </div>
    </InfoStyles>
  );
};

// EXPORTS
// --------------------------------------------------------
export default Info;
