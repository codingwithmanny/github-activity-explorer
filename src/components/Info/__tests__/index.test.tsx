// IMPORTS
// --------------------------------------------------------
import React from 'react';
import renderer from 'react-test-renderer';

// TESTING
// --------------------------------------------------------
import Info from '../';

// TEST
// --------------------------------------------------------
/**
 *
 */
test('Renders <Info /> Show Null', () => {
  const tree = renderer.create(<Info show={null} />).toJSON();
  expect(tree).toMatchSnapshot();
});

/**
 *
 */
test('Renders <Info /> Show True', () => {
  const tree = renderer.create(<Info show={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});

/**
 *
 */
test('Renders <Info /> Show False', () => {
  const tree = renderer.create(<Info show={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});
