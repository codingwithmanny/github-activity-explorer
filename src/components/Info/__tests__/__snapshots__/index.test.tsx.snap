// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Info /> Show False 1`] = `
<div
  className="sc-AxjAm woTqO"
>
  <div>
    <button
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="StyledIconBase-ea9ulj-0 jVnGjM"
        fill="none"
        focusable="false"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1={18}
          x2={6}
          y1={6}
          y2={18}
        />
        <line
          x1={6}
          x2={18}
          y1={6}
          y2={18}
        />
      </svg>
    </button>
    <p>
      <small>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 jVnGjM"
          fill="none"
          focusable="false"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 00-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0020 4.77 5.07 5.07 0 0019.91 1S18.73.65 16 2.48a13.38 13.38 0 00-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 005 4.77a5.44 5.44 0 00-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 009 18.13V22"
          />
        </svg>
        GitHub URL
      </small>
    </p>
    <p>
      <a
        href="undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        <code />
      </a>
    </p>
    <p>
      <small>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 jVnGjM"
          fill="none"
          focusable="false"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx={12}
            cy={12}
            r={4}
          />
          <line
            x1={1.05}
            x2={7}
            y1={12}
            y2={12}
          />
          <line
            x1={17.01}
            x2={22.96}
            y1={12}
            y2={12}
          />
        </svg>
        GitHub COMMIT
      </small>
    </p>
    <p>
      <a
        href="undefined/commit/undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        <code />
      </a>
    </p>
    <p>
      <small>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 jVnGjM"
          fill="none"
          focusable="false"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="14 9 9 4 4 9"
          />
          <path
            d="M20 20h-7a4 4 0 01-4-4V4"
          />
        </svg>
        Deployment
      </small>
    </p>
    <p>
      <a
        href="undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        <code />
      </a>
    </p>
    <p>
      <small>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 jVnGjM"
          fill="none"
          focusable="false"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 21v-2a4 4 0 00-4-4H8a4 4 0 00-4 4v2"
          />
          <circle
            cx={12}
            cy={7}
            r={4}
          />
        </svg>
        GitHub User
      </small>
    </p>
    <p>
      <a
        href="https://github.com/undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        <code>
          @
        </code>
      </a>
    </p>
  </div>
</div>
`;

exports[`Renders <Info /> Show Null 1`] = `
<div
  className="sc-AxjAm hHnPIK"
>
  <div>
    <button
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="StyledIconBase-ea9ulj-0 jVnGjM"
        fill="none"
        focusable="false"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1={18}
          x2={6}
          y1={6}
          y2={18}
        />
        <line
          x1={6}
          x2={18}
          y1={6}
          y2={18}
        />
      </svg>
    </button>
    <p>
      <small>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 jVnGjM"
          fill="none"
          focusable="false"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 00-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0020 4.77 5.07 5.07 0 0019.91 1S18.73.65 16 2.48a13.38 13.38 0 00-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 005 4.77a5.44 5.44 0 00-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 009 18.13V22"
          />
        </svg>
        GitHub URL
      </small>
    </p>
    <p>
      <a
        href="undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        <code />
      </a>
    </p>
    <p>
      <small>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 jVnGjM"
          fill="none"
          focusable="false"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx={12}
            cy={12}
            r={4}
          />
          <line
            x1={1.05}
            x2={7}
            y1={12}
            y2={12}
          />
          <line
            x1={17.01}
            x2={22.96}
            y1={12}
            y2={12}
          />
        </svg>
        GitHub COMMIT
      </small>
    </p>
    <p>
      <a
        href="undefined/commit/undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        <code />
      </a>
    </p>
    <p>
      <small>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 jVnGjM"
          fill="none"
          focusable="false"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="14 9 9 4 4 9"
          />
          <path
            d="M20 20h-7a4 4 0 01-4-4V4"
          />
        </svg>
        Deployment
      </small>
    </p>
    <p>
      <a
        href="undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        <code />
      </a>
    </p>
    <p>
      <small>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 jVnGjM"
          fill="none"
          focusable="false"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 21v-2a4 4 0 00-4-4H8a4 4 0 00-4 4v2"
          />
          <circle
            cx={12}
            cy={7}
            r={4}
          />
        </svg>
        GitHub User
      </small>
    </p>
    <p>
      <a
        href="https://github.com/undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        <code>
          @
        </code>
      </a>
    </p>
  </div>
</div>
`;

exports[`Renders <Info /> Show True 1`] = `
<div
  className="sc-AxjAm eHDdCK"
>
  <div>
    <button
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="StyledIconBase-ea9ulj-0 jVnGjM"
        fill="none"
        focusable="false"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1={18}
          x2={6}
          y1={6}
          y2={18}
        />
        <line
          x1={6}
          x2={18}
          y1={6}
          y2={18}
        />
      </svg>
    </button>
    <p>
      <small>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 jVnGjM"
          fill="none"
          focusable="false"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 00-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0020 4.77 5.07 5.07 0 0019.91 1S18.73.65 16 2.48a13.38 13.38 0 00-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 005 4.77a5.44 5.44 0 00-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 009 18.13V22"
          />
        </svg>
        GitHub URL
      </small>
    </p>
    <p>
      <a
        href="undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        <code />
      </a>
    </p>
    <p>
      <small>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 jVnGjM"
          fill="none"
          focusable="false"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx={12}
            cy={12}
            r={4}
          />
          <line
            x1={1.05}
            x2={7}
            y1={12}
            y2={12}
          />
          <line
            x1={17.01}
            x2={22.96}
            y1={12}
            y2={12}
          />
        </svg>
        GitHub COMMIT
      </small>
    </p>
    <p>
      <a
        href="undefined/commit/undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        <code />
      </a>
    </p>
    <p>
      <small>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 jVnGjM"
          fill="none"
          focusable="false"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="14 9 9 4 4 9"
          />
          <path
            d="M20 20h-7a4 4 0 01-4-4V4"
          />
        </svg>
        Deployment
      </small>
    </p>
    <p>
      <a
        href="undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        <code />
      </a>
    </p>
    <p>
      <small>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 jVnGjM"
          fill="none"
          focusable="false"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 21v-2a4 4 0 00-4-4H8a4 4 0 00-4 4v2"
          />
          <circle
            cx={12}
            cy={7}
            r={4}
          />
        </svg>
        GitHub User
      </small>
    </p>
    <p>
      <a
        href="https://github.com/undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        <code>
          @
        </code>
      </a>
    </p>
  </div>
</div>
`;
